bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/RefreshStatusReceiver.java

```diff
@@ -0,0 +1,107 @@
+ * Therefore it needs to apply the information it to the right thing.
+ *
+ * Discovery uses the same response code, so we must not refresh the status during discovery
+ *
+ * @author Stefan Höhn - Initial contribution
+ */
+@NonNullByDefault
+public class RefreshStatusReceiver implements Runnable {
+    final Logger logger = LoggerFactory.getLogger(RefreshStatusReceiver.class);
+    private static final Gson GSON = new Gson();
+
+    public RefreshStatusReceiver() {
+    }
+
+    @Override
+    public void run() {
+        if (GoveeDiscoveryService.isDiscoveryActive()) {
+            logger.debug("Not running refresh as Scan is currently active");
+        }
+
+        GoveeHandler.refreshJobRunning = true;
+        logger.trace("REFRESH: running refresh cycle for {} devices", GoveeHandler.THING_HANDLERS.size());
+
+        if (GoveeHandler.THING_HANDLERS.isEmpty()) {
+            return;
+        }
+
+        GoveeHandler thingHandler;
+
+        try (MulticastSocket socket = new MulticastSocket(GoveeHandler.RECEIVEFROMDEVICE_PORT)) {
+            byte[] buffer = new byte[10240];
+            DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
+            socket.setReuseAddress(true);
+            logger.debug("waiting for Status");
+            socket.receive(packet);
+
+            String response = new String(packet.getData()).trim();
+            String deviceIPAddress = packet.getAddress().toString().replace("/", "");
+            logger.trace("Response from {} = {}", deviceIPAddress, response);
+            logger.trace("received = {} from {}", response, deviceIPAddress);
+
+            thingHandler = GoveeHandler.THING_HANDLERS.get(deviceIPAddress);
+            if (thingHandler == null) {
+                logger.warn("thing Handler for {} couldn't be found.", deviceIPAddress);
+                return;
+            }
+
+            logger.debug("updating status for thing {} ", thingHandler.getThing().getLabel());
+
+            if (!response.isEmpty()) {
+                try {
+                    StatusResponse statusMessage = GSON.fromJson(response, StatusResponse.class);
+                    thingHandler.updateDeviceState(statusMessage);
+                } catch (JsonSyntaxException jse) {
+                    thingHandler.statusUpdate(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            jse.getMessage());
+                }
+            } else {
+                thingHandler.statusUpdate(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/offline.communication-error.empty-response");
+            }
+            if (!thingHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                thingHandler.statusUpdate(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            logger.error("exception when receiving status packet {}", Arrays.toString(e.getStackTrace()));
```

Comment:
Will the binding stop working at this point?

Reply:
No it doesn't it will still work. It will just ignore the situation.

Reply:
In that case, please reduce log level to at least warn. See https://www.openhab.org/docs/developer/guidelines.html#f-logging.